grafana-plugin/src/pages/chat-ops/parts/tabs/SlackSettings/SlackSettings.tsx

```diff
@@ -25,23 +25,54 @@ const cx = cn.bind(styles);
 
 interface SlackProps extends WithStoreProps {}
 
-interface SlackState {}
+interface SlackState {
+  showENVVariablesButton: boolean;
+}
 
 @observer
 class SlackSettings extends Component<SlackProps, SlackState> {
-  state: SlackState = {};
+  state: SlackState = {
+    showENVVariablesButton: false,
+  };
 
   componentDidMount() {
+    this.getSlackLiveSettings();
     this.update();
   }
 
+  handleOpenSlackInstructions = () => {
+    const { store } = this.props;
+    store.slackStore.installSlackIntegration();
+  };
+
   update = () => {
     const { store } = this.props;
 
     store.slackChannelStore.updateItems();
     store.slackStore.updateSlackSettings();
   };
 
+  getSlackLiveSettings = async () => {
+    const { store } = this.props;
+    const slackClientOAUTH = await store.globalSettingStore.getGlobalSettingItemByName('SLACK_CLIENT_OAUTH_ID');
+    const slackClientOAUTHSecret = await store.globalSettingStore.getGlobalSettingItemByName(
+      'SLACK_CLIENT_OAUTH_SECRET'
+    );
+    const slackRedirectHost = await store.globalSettingStore.getGlobalSettingItemByName(
+      'SLACK_INSTALL_RETURN_REDIRECT_HOST'
+    );
+    const slackSigningSecret = await store.globalSettingStore.getGlobalSettingItemByName('SLACK_SIGNING_SECRET');
+
```

Comment:
it would be nice not to request next global setting if previous has error, because one mistake is enough to set `showENVVariablesButton: true`

Reply:
Added only 1 request getAll and checked only results